tuning/utils/import_utils.py

```diff
@@ -15,8 +15,6 @@
 # Third Party
 from transformers.utils.import_utils import _is_package_available
 
-_is_aim_available = _is_package_available("aim")
 
-
-def is_aim_available():
-    return _is_aim_available
+def is_package_available(pkg):
+    return _is_package_available(pkg)
```

Comment:
Let's not do this, `_is_package_available` function from `transformers` is an expensive operation as hits the filesystem and can be unnecessary performance overhead if `is_package_available` is used widely in the code. Lets revert the changes.

Reply:
Not sure if what you mean is clear to me. The package check for `aim` or any other tracker should be done before using them so cannot be avoided.
Do you just mean that people should not be using `is_package_available` where it is not absolutely necessary? 

Shouldn't we rather document the expensive nature of this operation here so that anyone who is using this in code knows what they are calling.

Reply:
> The package check for aim or any other tracker should be done before using them so cannot be avoided.

I agree with you on this. 

> Do you just mean that people should not be using is_package_available where it is not absolutely necessary?

I did not mean this.

The previous code does a one time check for aim package and the function `is_aim_available` pulls from the cache for that process. Which is much more efficient than the current implementation. So please revert the current changes and use `is_aim_available` function in your code.

Reply:
The current code only registers tracks once and hence calls `_is_package_available` only once.

Reply:
Right, I am just worried the function is now made `is_package_available ` general purpose for any package and this can be used at multiple places going forward. Just to support my point, same design is followed by HuggingFace transformers library to overcome the obvious overhead. However, I am okay with your current change.

Reply:
Sure @kmehant the intent what you say is okay with me...I concur....maybe then I will just change this in another way to not have this problem